input-output-hk/lace
# fix(staking): LW-7746 prevent "next" button from refocus after cancelling changing pools

# Checklist

- [x] JIRA - LW-7746

---

## Proposed solution

Expose `focusTriggerAfterClose` property from the underlying ant design modal which prevents the "trigger", i.e. "next" button from being focused (and the undesired light purple border around it to be shown) after closing the `ChangingPreferencesModal`

ant design modal docs for reference: https://ant.design/components/modal

## Testing

1. Go to `Staking -> Browse pools`
2. select some pools from the list
3. click "Next" on the "portfolio bar"
4. on the modal that shows up (`ChangingPreferencesModal`), click "Cancel"
5. Verify that there is no border around the "Next" button

## Screenshots

How the "next" button looked like before this PR, if user cancelled the prompt":

![Screenshot 2023-08-22 at 12 21 09](https://github.com/input-output-hk/lace/assets/4980147/65545d8e-34ed-4cbf-b496-6b2543a4c33e)

How it looks now:

![Screenshot 2023-08-22 at 12 21 21](https://github.com/input-output-hk/lace/assets/4980147/cb110e22-13e7-44e2-9558-ac321a5aba66)



---
# Allure report
`allure-report-publisher` generated test report!



**smokeTests**: ✅ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/linux/chrome/1649/5951539127/index.html) for [a8c9bf1a](https://github.com/input-output-hk/lace/pull/427/commits/a8c9bf1a7e3d6ed1118ae6d5dc13b0f09bd36ad3)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 35     | 0      | 0       | 0     | 35    | ✅     |
